google_guest_agent/network/manager/systemd_networkd_linux_test.go

```diff
@@ -0,0 +1,548 @@
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						returnErr: true,
+					},
+				},
+			},
+			expectedRes: false,
+			expectErr:   true,
+			expectedErr: "error checking networkctl version: mock error version",
+		},
+		// networkctl is-active error.
+		{
+			name: "networkctl-is-active-error",
+			opts: testOpts{
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						version: 300,
+					},
+					isActiveErr: true,
+				},
+			},
+			expectedRes: false,
+			expectErr:   false,
+		},
+		// networkctl status error.
+		{
+			name: "networkctl-status-error",
+			opts: testOpts{
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						version: 300,
+					},
+					statusOpts: statusOpts{
+						returnErr: true,
+					},
+				},
+			},
+			expectedRes: false,
+			expectErr:   true,
+			expectedErr: "failed to check systemd-networkd network status: mock error status",
+		},
+		// networkctl status no networkctl key.
+		{
+			name: "networkctl-status-no-key",
+			opts: testOpts{
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						version: 300,
+					},
+					statusOpts: statusOpts{
+						returnValue: false,
+						hasKey:      false,
+					},
+				},
+			},
+			expectedRes: false,
+			expectErr:   true,
+			expectedErr: "could not determine interface state, one of [AdministrativeState SetupState] was not present",
+		},
+		// networkctl status interface is unmanaged.
+		{
+			name: "networkctl-status-unmanaged",
+			opts: testOpts{
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						version: 300,
+					},
+					statusOpts: statusOpts{
+						returnValue:   false,
+						hasKey:        true,
+						configuredKey: "AdministrativeState",
+					},
+				},
+			},
+			expectedRes: false,
+			expectErr:   false,
+		},
+		// networkctl status interface is managed. Whole method passes.
+		{
+			name: "pass",
+			opts: testOpts{
+				lookPathOpts: lookPathOpts{
+					returnValue: true,
+				},
+				runnerOpts: runnerOpts{
+					versionOpts: versionOpts{
+						version: 300,
+					},
+					statusOpts: statusOpts{
+						returnValue:   true,
+						hasKey:        true,
+						configuredKey: "SetupState",
+					},
+				},
+			},
+			expectedRes: true,
+			expectErr:   false,
+		},
+
+	}
+
+	for _, test := range tests {
+		t.Run(fmt.Sprintf("test-systemd-networkd-is-managing-%s", test.name), func(t *testing.T) {
+			ctx := context.Background()
+			systemdTestSetup(t, test.opts)
+
+			res, err := mockSystemd.IsManaging(ctx, "iface")
+
+			// Check expected errors.
+			if err != nil && !test.expectErr {
+				t.Fatalf("err returned when none expected: %v", err)
```

Comment:
I guess you want `t.Errorf()` here, we want to fail this test instance not all instances, right?